File: 11기 코딩테스트_230308/연습1/HasMatchedString/HasMatchedString_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.HasMatchedString.HasMatchedString import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        s = 'ka'
        target = 'kakao'

        # when
        result = solution(s, target)

        # then
        self.assertEqual(True, result)

    def test_2(self):
        # given
        s = ''
        target = 'naver'

        # when
        result = solution(s, target)

        # then
        self.assertEqual(True, result)

    def test_3(self):
        # given
        s = 'Goo'
        target = 'google'

        # when
        result = solution(s, target)

        # then
        self.assertEqual(False, result)

    def test_4(self):
        # given
        s = 'z'
        target = 'abcdefg'

        # when
        result = solution(s, target)

        # then
        self.assertEqual(False, result)

    def test_5(self):
        # given
        s = 'ObJect'
        target = 'ObJect'

        # when
        result = solution(s, target)

        # then
        self.assertEqual(True, result)

    def test_6(self):
        # given
        s = 'apple'
        target = ''

        # when
        result = solution(s, target)

        # then
        self.assertEqual(False, result)


if __name__ == '__main__':
    unittest.main()

File: 11기 코딩테스트_230308/연습1/HasMatchedString/HasMatchedString_test.js
import assert from 'assert'
import solution from './HasMatchedString.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const s = 'ka'
    const target = 'kakao'

    // when
    const result = solution(s, target)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const s = ''
    const target = 'naver'

    // when
    const result = solution(s, target)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const s = 'Goo'
    const target = 'google'

    // when
    const result = solution(s, target)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const s = 'z'
    const target = 'abcdefg'

    // when
    const result = solution(s, target)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const s = 'ObJect'
    const target = 'ObJect'

    // when
    const result = solution(s, target)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const s = 'apple'
    const target = ''

    // when
    const result = solution(s, target)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });
})
